airflow/cli/commands/task_command.py

```diff
@@ -226,7 +227,9 @@ def _run_task_by_executor(args, dag, ti):
         mark_success=args.mark_success,
         pickle_id=pickle_id,
         ignore_all_deps=args.ignore_all_dependencies,
-        ignore_depends_on_past=args.ignore_depends_on_past,
+        ignore_depends_on_past=(
+            args.ignore_depends_on_past or args.depends_on_past == PastDependenciesAction.IGNORE
+        ),
```

Comment:
Is the first part before the `or` ever possible? Since you set a default in the parser, this should always be set, is it not?

Reply:
The default value for the first part is still `False`, the goal of this `or` is to support the new configuration and maintain b/c for the existing argument until remove it:
- if  `--ignore-depends-on-past` is set -> ignore_depends_on_past will be `True` regardless the value of `--depends-on-past` (`True or <anything>`)
- if not:
  - and `--depends-on-past check` (default) ->  ignore_depends_on_past will be `False`  (`False or False`)
  - and `--depends-on-past ignore` ->  ignore_depends_on_past will be `True` (`False or True`)